src/modm/platform/uart/stm32/uart_hal_impl.hpp.in

```diff
@@ -33,14 +33,8 @@ modm::platform::{{ name }}::setParity(const Parity parity)
 	}
 %% endif
 
-	uint32_t flags = {{ peripheral }}->CR1;
-	flags &= ~(USART_CR1_PCE | USART_CR1_PS | USART_CR1_M);
-	flags |= static_cast<uint32_t>(parity);
-	if (parity != Parity::Disabled) {
-		// Parity Bit counts as 9th bit -> enable 9 data bits
-		flags |= USART_CR1_M;
-	}
-	{{ peripheral }}->CR1 = flags;
+	{{ peripheral }}->CR1 = ({{ peripheral }}->CR1 & ~(USART_CR1_PCE | USART_CR1_PS)) | static_cast<uint32_t>(parity);
+	setWordLength(parity == Parity::Enabled ? WordLength::Bit9 : WordLength::Bit8);
```

Comment:
This isn't a good API, since technically you can have parity on 7-bit data (like ASCII) and then you get a 8-bit word. I would prefer to remove this line here. What do you think?